add new commands QuickhlManualNext, QuickhlManualPrev

micchy326 opened the pull request:
Thank you for the great plugin.
This PR adds some jump commands.
- `<Plug>(quickhl-manual-next)`
- `<Plug>(quickhl-manual-prev)`
- `QuickhlManualNext`
- `QuickhlManualPrev`

I think these commands will improve the experience when reading large log files.

t9md commented:
Thank you for the great idea and feature you added.
I almost was about to merge immediately but I noticed several things.
Here are my thoughts, what do you think?

- Using `->` function call might break backward compatibility(I do not know how severe it is since I am not an active Vim user now).
- Instead of searching keywords from the added keywords, can't we get the exact positions from the locations which `matchadd()` function maintains and use that position as the destination to land? (just idea, I am not sure if it's possible). User can add regexp pattern when they call `QuichlManuall!`(NOTE ! at the end of cmd), does your code expect unescaped regex keyword comes as search pattern?
- To be explicit I want this command named `QuickhlManualGoToPrev`,  `QuickhlManualGoToNext`.

Overall I am happy to add this feature, but before that, I want to know your commend for the questions.
Thank you.

micchy326 commented:
Thank you for review.

I fixed these.
* "->" function is my mistake.
  It broke backward compatibility.
  I fixed it.

* I renamed commands follows.
  - <Plug>(quickhl-manual-go-to-next)
  - <Plug>(quickhl-manual-go-to-prev)
  - QuickhlManualGoToNext
  - QuickhlManualGoToPrev

> does your code expect unescaped regex keyword comes as search pattern?

Yes.
My understanding is that `color.pattern` always includes only escaped string.
So, necessary operation is just join by `\|`, I think.

t9md commented:
Thank you!
